examples/ensemble/plot_gradient_boosting_categorical.py

```diff
@@ -140,24 +143,14 @@
 # that will natively handle categorical features. This estimator will not treat
 # categorical features as ordered quantities.
 #
-# Since the :class:`~ensemble.HistGradientBoostingRegressor` requires category
-# values to be encoded in `[0, n_unique_categories - 1]`, we still rely on an
-# :class:`~preprocessing.OrdinalEncoder` to pre-process the data.
+# To benefit from this, the categorical features need to be encoded using the
+# pandas categorical dtype which we already did at the beginning of this
```

Comment:
The way this is currently phrased suggests that one *must* have pandas for categorical handling to work. Perhaps we can be clearer that this is just one option (the other option being the usage of OrdinalEncoder)?


```suggestion
# To benefit from this, one option is to encode the categorical features using the
# pandas categorical dtype which we already did at the beginning of this
```

and then below (see other comment)

> Another option is to use an ordinal encoder [...]